### DISCUSSION TITLE
Explain box.runtime.info()

### DISCUSSION THREAD

**Discussion by DifferentialOrange** [tarantool/tarantool#7106] [Q&A] [answered]
There are a few lines about box.runtime metrics in doc:

[box.runtime.info()](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_slab/runtime_info/):
> lua is the heap size of the Lua garbage collector;
> used is the current memory size used by Lua.

[Metrics runtime](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_slab/runtime_info/):
> tnt_runtime_lua Lua garbage collector size in bytes
> tnt_runtime_used  Number of bytes used for the Lua runtime

`tnt_runtime_lua` seems to be the same as (or something close to) `box.info.memory().lua`:
![image](https://user-images.githubusercontent.com/20455996/166659184-fc3bf08e-774d-42e6-9223-228cd6196586.png)

At the same time, `tnt_runtime_used` seems monotonic and always greater than `tnt_runtime_lua`
![9d23-6c57-e251-ceb6](https://user-images.githubusercontent.com/20455996/166659618-2c024330-9d04-4948-b65d-4a9dfaac0632.png)

It's hard to notice any similarities between slab and runtime arena:
![image](https://user-images.githubusercontent.com/20455996/166659673-0105ea01-40e4-40f8-be5a-724eeda953e9.png)

So the question is: what is `box.info.memory().used` (`tnt_runtime_used`) and how we should monitor it? Is `tnt_runtime_used` reaching 2 Gb Lua memory limit is an alert condition or not?

**Comment by Totktonada** [comment, marked as answer]
See also https://github.com/tarantool/doc/issues/2911.

**Comment by DifferentialOrange** [comment]
The exact answer I came up with based on https://github.com/tarantool/doc/issues/2911 is

> `tnt_runtime_used` is memory used by runtime arena. Runtime arena stores network buffers, tuples created with `box.tuple.new` and other objects allocated by application not covered by basic Lua mechanisms (spaces data and indexes are not included here, see memtx/vinyl arena).
